rqt_controller_manager/rqt_controller_manager/controller_manager.py

```diff
@@ -0,0 +1,425 @@
+                self._activate_controller(ctrl.name)
+
+    def _on_ctrl_info(self, index):
+        popup = self._popup_widget
+
+        ctrl = self._controllers[index.row()]
+        popup.ctrl_name.setText(ctrl.name)
+        popup.ctrl_type.setText(ctrl.type)
+
+        res_model = QStandardItemModel()
+        model_root = QStandardItem('Claimed Interfaces')
+        res_model.appendRow(model_root)
+        for claimed_interface in ctrl.claimed_interfaces:
+            hw_iface_item = QStandardItem(claimed_interface)
+            model_root.appendRow(hw_iface_item)
+
+        popup.resource_tree.setModel(res_model)
+        popup.resource_tree.setItemDelegate(FontDelegate(popup.resource_tree))
+        popup.resource_tree.expandAll()
+        popup.move(QCursor.pos())
+        popup.show()
+
+    def _on_header_menu(self, pos):
+        header = self._widget.table_view.horizontalHeader()
+
+        # Show context menu
+        menu = QMenu(self._widget.table_view)
+        action_toggle_auto_resize = menu.addAction('Toggle Auto-Resize')
+        action = menu.exec_(header.mapToGlobal(pos))
+
+        # Evaluate user action
+        if action is action_toggle_auto_resize:
+            if header.resizeMode(0) == QHeaderView.ResizeToContents:
+                header.setSectionResizeMode(QHeaderView.Interactive)
+            else:
+                header.setSectionResizeMode(QHeaderView.ResizeToContents)
+
+    def _activate_controller(self, name):
+        switch_controllers(
+            node=self._node,
+            controller_manager_name=self._cm_name,
+            deactivate_controllers=[],
+            activate_controllers=[name],
+            strict=SwitchController.Request.STRICT,
+            activate_asap=False,
+            timeout=0.3
+        )
+
+    def _deactivate_controller(self, name):
+        switch_controllers(
+            node=self._node,
+            controller_manager_name=self._cm_name,
+            deactivate_controllers=[name],
+            activate_controllers=[],
+            strict=SwitchController.Request.STRICT,
+            activate_asap=False,
+            timeout=0.3
+        )
+
+
+class ControllerTable(QAbstractTableModel):
+    """
+    Model containing controller information for tabular display.
+
+    The model allows display of basic read-only information like controller
+    name and state.
+    """
+
+    def __init__(self, controller_info,  icons, parent=None):
+        QAbstractTableModel.__init__(self, parent)
+        self._data = controller_info
+        self._icons = icons
+
+    def rowCount(self, parent):
+        return len(self._data)
+
+    def columnCount(self, parent):
+        return 2
+
+    def headerData(self, col, orientation, role):
+        if orientation == Qt.Horizontal and role == Qt.DisplayRole:
+            if col == 0:
+                return 'controller'
+            elif col == 1:
+                return 'state'
+        else:
+            return None
```

Comment:
Perhaps, we can reduce the complexity of this function with something like this
```suggestion
    def headerData(self, col, orientation, role):
        if orientation != Qt.Horizontal or role != Qt.DisplayRole:
            return None
        if col == 0:
            return 'controller'
        elif col == 1:
            return 'state'
```

The weird thing is that I don't know if we can have another value for `col`, otherwise we can do a `return 'state' if col == 1 else 'controller'`